triqs/gfs/gf/defs.hpp

```diff
@@ -91,6 +91,24 @@ namespace triqs {
 
     template <typename G> TRIQS_DEPRECATED("use X.target_shape() instead") auto get_target_shape(G const &g) { return g.target_shape(); }
 
+    /**----------------------------------------------------------
+  *  @return  - A new array of zeros of 0 if Gf is matrix/tensor valued.
+  *           - 0 if Gf is scalar valued.
+  *--------------------------------------------------------*/
+
+    template <typename G> auto make_zero_return(G const &g) {
```

Comment:
Do we really want a function for this or shouldn't we just write
```c++
triqs::arrays::zeros(g.target_shape())
```
instead?

Reply:
It will not work for the scalar case.
But yes, we can simplify the implementation.

Reply:
Couldn't we even make that work for the scalar case?